### PR TITLE
add real-time data to metrics-summary

### PR DESCRIPTION
**Pull request by scrajesh**
When deploying a new app, they NEED real-time data to monitor for bad versions.  This usually occurs in the first 10 minutes or so.

Taking  the summation of the last time it's summed, query the database for the remaining real-time data. This query should be quick because it is time-bound. 

If the last summed time is > than 1 day, query only for 1 day's worth of data.

### CODE REVIEW

File: electrode-ota-server-model-app/src/app.js [hunk 1]
```diff
@@ -523,7 +580,29 @@ export default (options, dao, upload, logger) => {
...
+                }
+                summary[k].active += latestSummary[k].active;
+                summary[k].downloaded += latestSummary[k].downloaded;
+                summary[k].failed += latestSummary[k].failed;
+                summary[k].installed += latestSummary[k].installed;
+              }
+            }
+
+            return summary;
+          });
```
Comment (datvong-wm) [inline electrode-ota-server-model-app/src/app.js]: can u this in another function?

File: electrode-ota-server-model-app/src/app.js [hunk 2]
```diff
@@ -523,7 +580,29 @@ export default (options, dao, upload, logger) => {
...
       return this.getMetricsFromCache(deployment).then(result => {
         if (result && result.summaryJson) {
-          return JSON.parse(result.summaryJson);
+          const summary = JSON.parse(result.summaryJson);
+          // fetch latest metrics data
+          return dao.metricsByStatusAndTime(
+            deployment.key, new Date(result.lastRunTimeUTC), new Date(Date.now())
+          ).then((metrics = []) => {
+            const { label } = deployment.package || {};
+            logger.info({ deployment: deployment.name }, "fetched latest metrics by status and time");
```
Comment (datvong-wm) [inline electrode-ota-server-model-app/src/app.js]: do we need this logger?

File: electrode-ota-server-model-app/src/app.js [hunk 3]
```diff
@@ -523,7 +580,29 @@ export default (options, dao, upload, logger) => {
...
+            // cleanup and merge the results
+            const latestSummary = doSummarize(metrics, label);
+            if (isValidObject(latestSummary)) {
```
[the remark this answers is not in the document]
Comment (scrajesh) [reply electrode-ota-server-model-app/src/app.js]: If there're no real-time data. i.e if one requests for /metrics immediately after a Summarize run.

File: electrode-ota-server-model-app/src/app.js (hunk 2 above)
Comment (scrajesh) [reply electrode-ota-server-model-app/src/app.js]: i can remove it if not required.

File: electrode-ota-server-model-app/src/app.js (hunk 1 above)
Comment (scrajesh) [reply electrode-ota-server-model-app/src/app.js]: u mean to a different file ?

Comment (datvong-wm) [reply electrode-ota-server-model-app/src/app.js]: another function in same file.

File: electrode-ota-server-model-app/src/app.js (hunk 2 above)
Comment (scrajesh) [reply electrode-ota-server-model-app/src/app.js]: removed it.

File: electrode-ota-server-model-app/src/app.js (hunk 1 above)
Comment (scrajesh) [reply electrode-ota-server-model-app/src/app.js]: moved the metrics merger to as separate method.

**Pull request merged**